shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/create/CreateTrafficRuleHandler.java

```diff
@@ -0,0 +1,131 @@
+        }
+        Collection<String> invalidAlgorithmNames = getInvalidAlgorithmNames(sqlStatement.getSegments());
+        DistSQLException.predictionThrow(invalidAlgorithmNames.isEmpty(), new InvalidAlgorithmConfigurationException("traffic", invalidAlgorithmNames));
+    }
+    
+    private Collection<String> getInvalidAlgorithmNames(final Collection<TrafficRuleSegment> segments) {
+        Collection<String> result = new ArrayList<>(segments.size());
+        sqlStatement.getSegments().forEach(each -> {
+            if (!TypedSPIRegistry.findRegisteredService(TrafficAlgorithm.class, each.getAlgorithm().getName(), new Properties()).isPresent()) {
+                result.add(each.getAlgorithm().getName());
+            }
+            if (!TypedSPIRegistry.findRegisteredService(TrafficLoadBalanceAlgorithm.class, each.getLoadBalancer().getName(), new Properties()).isPresent()) {
+                result.add(each.getLoadBalancer().getName());
+            }
+        });
+        return result;
+    }
+    
+    private TrafficRuleConfiguration createTrafficRuleConfiguration(final CreateTrafficRuleStatement sqlStatement) {
+        TrafficRuleConfiguration result = new TrafficRuleConfiguration();
+        sqlStatement.getSegments().forEach(each -> setConfigurationData(result, each));
+        return result;
+    }
+    
+    private void setConfigurationData(final TrafficRuleConfiguration result, final TrafficRuleSegment each) {
+        ShardingSphereAlgorithmConfiguration trafficAlgorithm = createAlgorithmConfiguration(each.getAlgorithm());
+        ShardingSphereAlgorithmConfiguration loadBalancer = createAlgorithmConfiguration(each.getLoadBalancer());
+        String trafficAlgorithmName = createAlgorithmName(each.getName(), trafficAlgorithm);
+        String loadBalancerName = createAlgorithmName(each.getName(), loadBalancer);
+        TrafficStrategyConfiguration trafficStrategy = createTrafficStrategy(each, trafficAlgorithmName, loadBalancerName);
+        result.getTrafficStrategies().add(trafficStrategy);
+        result.getTrafficAlgorithms().put(trafficAlgorithmName, trafficAlgorithm);
+        result.getLoadBalancers().put(loadBalancerName, loadBalancer);
+    }
+    
+    private ShardingSphereAlgorithmConfiguration createAlgorithmConfiguration(final AlgorithmSegment segment) {
+        return new ShardingSphereAlgorithmConfiguration(segment.getName(), segment.getProps());
+    }
+    
+    private TrafficStrategyConfiguration createTrafficStrategy(final TrafficRuleSegment trafficRuleSegment, final String trafficAlgorithmName, final String loadBalancerName) {
+        return new TrafficStrategyConfiguration(trafficRuleSegment.getName(), trafficRuleSegment.getLabels(), trafficAlgorithmName, loadBalancerName);
+    }
+    
+    private void updateToRepository(final TrafficRuleConfiguration newTrafficRuleConfiguration, final Optional<TrafficRuleConfiguration> trafficRuleConfiguration) {
```

Comment:
Refer to RuleDefinitionCreateUpdater, use `toBeCreatedRuleConfig` and `currentRuleConfig` to name the params.